lib/matplotlib/axes/_axes.py

```diff
@@ -7369,12 +7396,44 @@ def _kde_method(X, coords):
             kde = mlab.GaussianKDE(X, bw_method)
             return kde.evaluate(coords)
 
-        vpstats = cbook.violin_stats(dataset, _kde_method, points=points)
-        return self.violin(vpstats, positions=positions, vert=vert,
+        def _resolve_duplicate_alias(func_obj_list):
+            unique_alias_set = set()
+            for func_obj in func_obj_list:
+                while func_obj.alias in unique_alias_set:
+                    func_obj.alias += 'x'
+                unique_alias_set.add(func_obj.alias)
```

Comment:
`unique_alias_set` isn't used or defined anywhere else, so I'm not sure why we even need the aliases. Why can the user just pass a list of functions that accept an array as input and returns an array or scalar, and draw lines at each value? For instance:

```python
from functools import partial
stat_fxn = [partial(np.percentile, q=[5, 25, 50, 75, 95])]
# OR
stat_fxns = [np.mean, np.median, partial(np.percentile, q=95)]
```

should work.